ts/mains/main_node.ts

```diff
@@ -1023,6 +1023,60 @@ ipc.on('get-start-in-tray', event => {
   }
 });
 
+ipc.on('set-auto-start-enabled', (event, newValue) => {
+  try {
+    userConfig.set('autoStart', newValue);
+
+    // Set the login item settings based on the platform
+    if (process.platform === 'darwin') {
+      // macOS
+      app.setLoginItemSettings({
+        openAtLogin: newValue,
+        openAsHidden: false,
+      });
+    } else if (process.platform === 'win32') {
+      // Windows - For Squirrel-based apps, we need to handle the stub launcher - https://www.electronjs.org/docs/latest/api/app/#appsetloginitemsettingssettings-macos-windows
+      const appFolder = path.dirname(process.execPath);
+      const ourExeName = path.basename(process.execPath);
+      const stubLauncher = path.resolve(appFolder, '..', ourExeName);
+
+      app.setLoginItemSettings({
+        openAtLogin: newValue,
+        path: stubLauncher,
+        args: [],
+      });
+    } else {
+      // Linux and other platforms - basic support
+      app.setLoginItemSettings({
+        openAtLogin: newValue,
+      });
+    }
+
+    event.sender.send('set-auto-start-enabled-response', null);
+  } catch (e) {
+    event.sender.send('set-auto-start-enabled-response', e);
+  }
+});
+
+ipc.on('get-auto-start-enabled', event => {
+  try {
+    const configVal = userConfig.get('autoStart');
+
+    if (configVal !== undefined) {
+      event.sender.send('get-auto-start-enabled-response', configVal);
+      return;
+    }
+
+    const loginSettings = app.getLoginItemSettings();
+    const isEnabled = loginSettings.openAtLogin;
+
+    userConfig.set('autoStart', isEnabled);
```

Comment:
same comment, if `autoStart` is not required to be set and `loginSettings.openAtLogin` can be our source of truth, we should only rely on it